packages/code-studio/src/main/AppMainContainer.tsx

```diff
@@ -706,11 +709,73 @@ export class AppMainContainer extends Component<
     });
   });
 
-  render(): ReactElement {
-    const { activeTool, plugins, user, workspace, serverConfigValues } =
-      this.props;
+  handleHomeClick(): void {
+    this.handleTabSelect(DEFAULT_DASHBOARD_ID);
+  }
+
+  handleTabSelect(tabId: string): void {
+    this.setState({ activeTabKey: tabId });
+  }
+
+  handleTabReorder(from: number, to: number): void {
+    this.setState(({ tabs: oldTabs }) => {
+      const newTabs = [...oldTabs];
+      const [t] = newTabs.splice(from, 1);
+      newTabs.splice(to, 0, t);
+      return { tabs: newTabs };
+    });
+  }
+
+  handleTabClose(tabId: string): void {
+    const { setDashboardData } = this.props;
+    // TODO: Figure out how to remove the dashboard data
+    // without updates after this recreating some dashboard data
```

Comment:
Not sure I follow this TODO - what's wrong with this?

Reply:
So right now when you close a dashboard, we null out the dashboard data, but then other parts of the app emit stuff like the `closed` array which recreates dashboard data.

If we track dashboards via a different property then that can be fine. But since we have no way of the user managing dashboards, we need to clean them up at some point. Otherwise the dashboard data grows unbounded with unreachable dashboards

Reply:
See my other comment about remove `closed` array from the workspace data... does this TODO still need to be followed up on? File a separate ticket for it?

Reply:
I've removed the todo and the call to `setDashboardData`. Will leave this for a separate persistence PR